### ISSUE TITLE
Bug: ESLint thinks variables that are destructured and then spread are undefined

### ISSUE THREAD

**Issue by janhesters** [eslint/eslint#12564]
**Tell us about your environment**

* **ESLint Version:** 6.6.0
* **Node Version:** 12.13.0
* **npm Version:** 6.12

**What parser (default, Babel-ESLint, etc.) are you using?**
Jest with React Native, so probably default or Babel? 🤔 

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```json
{
  "extends": "universe/native",
  "plugins": [
    "simple-import-sort",
    "react-hooks"
  ],
  "rules": {
    "simple-import-sort/sort": "error",
    "import/order": "off",
    "react-hooks/rules-of-hooks": "error",
    "react-hooks/exhaustive-deps": "warn"
  },
  "settings": {
    "react": {
      "version": "detect"
    }
  }
}
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
const setPaymentID = ({ paymentID = '' } = {}) => ({
  type: setPaymentID.type,
  payload: { paymentID },
});
setPaymentID.type = 'ORDERS/setPaymentID';

const reducer = (state = initial, { type, payload } = {}) => {
  switch (type) {
    case setPaymentID.type:
      return { ...state, ...({ paymentID } = payload) }; // The bug happens here!
    // ... more cases
    default:
      return state;
  }
};
```

I ran:

```bash
yarn lint
```

which executes:

```json
"lint": "eslint --ignore-path .gitignore .",
```

**What did you expect to happen?**
I expected ESLint _**NOT**_ to complain but be fine with my code because it is valid JavaScript and there is nothing undefined.

**What actually happened? Please include the actual, raw output from ESLint.**
ESLint complained and said: 

```bash
'paymentID' is not defined. eslint(no-undef)
```

Even though it's just destructured and spread in one go. So it is defined and used.

![Screenshot 2019-11-14 at 08 48 00](https://user-images.githubusercontent.com/31096420/68836808-7b916a80-06bb-11ea-93d2-96782eced723.png)

**Comment by ljharb** [comment]
In this case, you're not declaring `paymentID` with `const` or `let` or `var`, so it definitely should be warning.

Destructuring should be done in a separate line to create a variable, and then you can use the variable. (and in this case, you can do `...payload.paymentID` anyways)

**Comment by janhesters** [comment]
@ljharb Hmm, okay 😕 Thank you!
